libs/lib_parsing/Parsed_int.mli

```diff
@@ -0,0 +1,42 @@
+(* Why parsed integers?
+ * This is a library for "parsed ints", which are a type that correspond to
+ * literal integers that have been parsed from source text.
+
+ * Integers have many obvious uses, such as counters, indices, etc. These can
+ * be conflated with the OCaml representation of integers in a target program,
+ * which are often not meant to be manipulated, but are merely symbols.
+
+ * As such, it is somewhat useful to have a separate notion of integers which are
+ * concrete in the target text, and integers which we make up for our internal
+ * purposes.
+ *)
+
+type t [@@deriving hash, show, eq]
+
+(* creators *)
+val of_int : int -> t
+val of_int64 : int64 -> t
+val parse : string * Tok.t -> t
```

Comment:
I would put parse and parse_c_octal at the top and says (* Those are the 2 main functions you should use to build parsed_int *)